src/App.js

```diff
@@ -14,11 +14,110 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selected: [],
+  };
+
+  clearGoods = () => {
+    this.setState({
+      selected: [],
+    });
+  }
+
+  addGood = (good) => {
+    this.setState(prevState => ({
+      selected: [...prevState.selected, good],
+    }));
+  }
+
+  removeGood = (goodToRemove) => {
+    if (this.state.selected.includes(goodToRemove)) {
+      this.setState(prevState => ({
+        selected: prevState.selected.filter(
+          good => good !== goodToRemove,
+        ),
+      }));
+    }
+  }
+
+  listRenderer() {
+    const goods = this.state.selected;
+
+    if (!goods.length) {
+      return `No goods are selected`;
+    }
+
+    if (goods.length === 1) {
+      return `${goods} is selected`;
+    }
+
+    if (goods.length === 2) {
+      return `${goods[0]} and ${goods[1]} are selected`;
+    }
+
+    return `${goods.slice(0, -1).join(`, `)}`
+    + ` and ${goods.slice(-1)} are selected`;
+  }
+
+  render() {
+    const { selected } = this.state;
+
+    return (
+      <div className="card">
+        <div className="card__goods">
+          <h2>
+            Selected goods:
+            {` ${this.listRenderer()}`}
+          </h2>
+          <p>
+            Number of goods selected:
+            <span>
+              {` ${selected.length}`}
+            </span>
+          </p>
+          <button
+            type="button"
+            className="card__button"
+            onClick={() => this.clearGoods()}
```

Comment:
якщо функція clearGoods не приймає параметрів то стрілочну функцію не потрібно писати
```suggestion
            onClick={this.clearGoods}
```